core/src/main/java/org/mule/runtime/core/processor/BlockMessageProcessor.java

```diff
@@ -74,17 +81,37 @@ protected TransactionFactory getTransactionFactory() {
     return new DelegateTransactionFactory();
   }
 
-  protected MuleTransactionConfig createTransactionConfig(String action) {
+  protected MuleTransactionConfig createTransactionConfig(String action, TransactionType type) {
     MuleTransactionConfig transactionConfig = new MuleTransactionConfig();
     transactionConfig.setActionAsString(action);
+    transactionConfig.setFactory(lookUpTransactionFactory(type));
     return transactionConfig;
   }
 
+  private TransactionFactory lookUpTransactionFactory(TransactionType type) {
```

Comment:
Just to make sure there are no incompatibilities if the factory bean is eventually removed